elastalert/elastalert.py

```diff
@@ -460,7 +460,8 @@ def run_rule(self, rule, endtime, starttime=None):
                     key = '.' + str(match[rule['query_key']])
                 except KeyError:
                     # Some matches may not have a query key
-                    key = ''
+                    # Use a special token for these to not clobber all alerts
+                    key = '._missing'
```

Comment:
Make sure this is documented somewhere, in the possible event of namespace conflicts (ie, a real key named "._missing")